### PR TITLE
DOCS-5877 Add DORA Metrics Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Add documentation for DORA Metrics private beta release. Can be reviewed and merged in on Monday (Nov 20)!

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

**Comment by AdrianLC** [comment]
hey @alai97 thanks a lot for the help with this 😁 

How do you want to collaborate on this? Should I commit more content to your branch directly? or would you rather merge this to an integration branch and I can add more PRs targeting it before we do the final merge to master?

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+- `service`
+- `finished_at`
+
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: https://cloud.google.com/blog/products/devops-sre/using-the-four-keys-to-measure-your-devops-performance
+[2]: /api/latest/continuous_integration/TBD
```
Comment (alai97) [inline content/en/continuous_integration/dora_metrics/_index.md:97]:
```suggestion
[2]: /api/latest/dora-metrics
```

**Review by kassenq — APPROVED** [review]

**Review by AdrianLC — APPROVED** [review]

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
+---
+title: DORA Metrics
+kind: documentation
+description: Learn how to use DORA metrics to improve and measure software development.
+is_beta: true
+further_reading:
+- link: "/continuous_integration/pipelines"
+  tag: "Documentation"
+  text: "Learn about Pipeline Visibility"
+- link: "https://github.com/DataDog/datadog-ci"
```
Comment (alai97) [inline content/en/continuous_integration/dora_metrics/_index.md:10]:
```suggestion
- link: "/tracing/service_catalog"
  tag: "Documentation"
  text: "Learn about the Service Catalog"
- link: "https://github.com/DataDog/datadog-ci"
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+  tag: "Source Code"
+  text: "Learn about the `datadog-ci` CLI tool"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+{{< callout url="https://forms.gle/Eqq6uXfGjYxmqpjDA" header="false" >}}
+DORA Metrics are in private beta for <a href="https://docs.datadoghq.com/continuous_integration/pipelines/">CI Pipeline Visibility</a> customers. To request access, complete the form.
```
Comment (kassenq) [inline content/en/continuous_integration/dora_metrics/_index.md:23]: "DORA Metrics are in private beta for CI Pipeline Visibility, [Test Visibility](https://docs.datadoghq.com/continuous_integration/tests/), and/or [Static Analysis](https://docs.datadoghq.com/continuous_integration/static_analysis/?tab=circleciorbs) customers."

**Review by kassenq — CHANGES_REQUESTED** [review]

File: content/en/continuous_integration/dora_metrics/_index.md (hunk 3 above)
Comment (AdrianLC) [inline content/en/continuous_integration/dora_metrics/_index.md:23]:
```suggestion
DORA Metrics are in private beta for <a href="https://docs.datadoghq.com/continuous_integration/pipelines/">CI Pipeline Visibility</a>, <a href="https://docs.datadoghq.com/continuous_integration/tests/">Test Visibility</a> and/or <a href="https://docs.datadoghq.com/continuous_integration/static_analysis/">Static Analysis</a> customers. To request access, complete the form.
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+  tag: "Source Code"
+  text: "Learn about the `datadog-ci` CLI tool"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+{{< callout url="https://forms.gle/Eqq6uXfGjYxmqpjDA" header="false" >}}
+DORA Metrics are in private beta for <a href="https://docs.datadoghq.com/continuous_integration/pipelines/">CI Pipeline Visibility</a>, <a href="https://docs.datadoghq.com/continuous_integration/tests/">Test Visibility</a> and/or <a href="https://docs.datadoghq.com/continuous_integration/static_analysis/">Static Analysis</a> customers. To request access, complete the form.
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:23]:
```suggestion
DORA Metrics are in private beta for <a href="https://docs.datadoghq.com/continuous_integration/pipelines/">CI Pipeline Visibility</a>, <a href="https://docs.datadoghq.com/continuous_integration/tests/">Test Visibility</a> and <a href="https://docs.datadoghq.com/continuous_integration/static_analysis/">Static Analysis</a> customers. To request access, complete the form.
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+{{% /tab %}}
+
+{{% tab "datadog-ci CLI" %}}
+
+The [`datadog-ci`][1] CLI tool provides a shortcut to send the deployments within CI.
+
+For the following example, set the `DD_SITE` environment variable to {{< region-param key="dd_site" code="true" >}}:
+```shell
+export DD_BETA_COMMANDS_ENABLED=1
+export DD_SITE="DD_SITE"
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:104]:
```suggestion
export DD_SITE="<DD_SITE>"
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+export DD_API_KEY="api-key"
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:105]:
```suggestion
export DD_API_KEY="<API_KEY>"
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+Reporting commit 007f7f466e035b052415134600ea899693e7bb34 from repository git@github.com:organization/example-repository.git.
+180 tracked file paths will be reported.
+✅  Handled in 0.077 seconds.
+```
+
+[1]: https://github.com/DataDog/datadog-ci/tree/master/src/commands/git-metadata
+{{% /tab %}}
+{{< /tabs >}}
+
+#### Filter git commits in monorepos
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:177]:
```suggestion
#### Filter Git commits in monorepos
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+
+If you are using a monorepo (building several services from the same git repository), not every git commit affects the lead time of all services.
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:179]:
```suggestion
If you are using a monorepo (building several services from the same Git repository), not every Git commit affects the lead time of all services.
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+
+To filter the commits measured to only the ones that affect the service, specify the source code glob file path patterns in the [Service definition][6].
+
+If the Service definition contains a **full** GitHub URL to the application folder, a single path pattern is automatically used.
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:183]:
```suggestion
If the service definition contains a **full** GitHub URL to the application folder, a single path pattern is automatically used.
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+
+```yaml
+extensions:
+  dora_metrics:
+    source_patterns:
+      - src/apps/shopist/**
+      - src/libs/utils/**
+```
+
+DORA Metrics for the service `shopist` only consider the git commits that include changes within `src/apps/shopist/**` or `src/libs/utils/**`.
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:207]:
```suggestion
DORA Metrics for the service `shopist` only consider the Git commits that include changes within `src/apps/shopist/**` or `src/libs/utils/**`.
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+datadog-ci dora deployment --service shopist --env prod \
+    --started-at $deploy_start --finished-at `date +%s` \
+    --git-repository-url "https://github.com/organization/example-repository" \
+    --git-commit-sha 66adc9350f2cc9b250b69abddab733dd55e1a588
+```
+
+You can optionally add the following parameters:
+  - `finished-at` is automatically set to now if not provided.
+  - `env`
+  - `git-repository-url` and `git-commit-sha` can be omitted if the deployment CI job is running on the exact same git checkout that has been deployed.
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:118]:
```suggestion
  - `git-repository-url` and `git-commit-sha` can be omitted if the deployment CI job is running on the exact same Git checkout that has been deployed.
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 12]
```diff
@@ -0,0 +1,292 @@
...
+  - `skip-git` disables the git details ([change lead time](#lead-time-for-changes) is not available).
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:119]:
```suggestion
  - `skip-git` disables the Git details ([change lead time](#lead-time-for-changes) is not available).
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 13]
```diff
@@ -0,0 +1,292 @@
...
+
+You can access and visualize your DORA metrics and filter them by team, service, repository, environment, and time period on the [DORA Metrics page][8].
+
+Use the information on this page to identify improvements or regressions for each metric, visualize changes, and compare trends over time. DORA metrics can be exported to dashboards and alerted on using [metric monitors][9].
+
+### Limitations
+
+- Deployment and incident events must be sent as soon as possible. Events for which the `started_at` timestamp is 1 hour older than the current time are not accepted.
+- Deployments or incidents of the same service cannot occur at the same second.
+- For [change lead time](#lead-time-for-changes), the retention of git metadata is 1 month. Commits older that 1 month are not accounted for.
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:278]:
```suggestion
- For [change lead time](#lead-time-for-changes), the retention of Git metadata is 1 month. Commits older that 1 month are not accounted for.
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 14]
```diff
@@ -0,0 +1,292 @@
...
+further_reading:
+- link: "/continuous_integration/pipelines"
+  tag: "Documentation"
+  text: "Learn about Pipeline Visibility"
+- link: "/tracing/service_catalog"
+  tag: "Documentation"
+  text: "Learn about the Service Catalog"
+- link: "https://github.com/DataDog/datadog-ci"
+  tag: "Source Code"
+  text: "Learn about the `datadog-ci` CLI tool"
```
Comment (jhgilbert) [inline content/en/continuous_integration/dora_metrics/_index.md:15]:
```suggestion
  text: "Learn about the datadog-ci CLI tool"
```
Backticks don't work in Further Reading, they just show up literally. Not sure how you feel about that -- you can leave them in if the visible backticks are better than nothing

**Review by jhgilbert — APPROVED** [review]

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 15]
```diff
@@ -0,0 +1,292 @@
+---
+title: DORA Metrics
+kind: documentation
+description: Learn how to use DORA metrics to improve and measure software development.
+is_beta: true
+further_reading:
+- link: "/continuous_integration/pipelines"
+  tag: "Documentation"
+  text: "Learn about Pipeline Visibility"
+- link: "/tracing/service_catalog"
```
Comment (alai97) [inline content/en/continuous_integration/dora_metrics/_index.md:10]:
```suggestion
- link: "/continuous_integration/static_analysis"
  tag: "Documentation"
  text: "Learn about Static Analysis"
- link: "/tracing/service_catalog"
```

File: content/en/continuous_integration/dora_metrics/_index.md [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+  tag: "Source Code"
+  text: "Learn about the datadog-ci CLI tool"
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+{{< callout url="https://forms.gle/Eqq6uXfGjYxmqpjDA" header="false" >}}
+DORA Metrics are in private beta for <a href="https://docs.datadoghq.com/continuous_integration/pipelines/">CI Pipeline Visibility</a>, <a href="https://docs.datadoghq.com/continuous_integration/tests/">Test Visibility</a> and <a href="https://docs.datadoghq.com/continuous_integration/static_analysis/">Static Analysis</a> customers. To request access, complete the form.
```
Comment (alai97) [inline content/en/continuous_integration/dora_metrics/_index.md:26]:
```suggestion
DORA Metrics are in private beta for <a href="https://docs.datadoghq.com/continuous_integration/pipelines/">CI Pipeline Visibility</a>, <a href="https://docs.datadoghq.com/continuous_integration/tests/">Test Visibility</a>, and <a href="https://docs.datadoghq.com/continuous_integration/static_analysis/">Static Analysis</a> customers. To request access, complete the form.
```

**Review by kassenq — APPROVED** [review]

**Pull request merged**